File: StudyGamePrograming06re/Dice.cpp
#include "Dice.h"
#include "MeshComponent.h"
#include "Game.h"
#include "Renderer.h"
#include "MoveComponent.h"

Dice::Dice(Game* game) :Actor(game)
{
	SetPosition(Vector3(200.0f, 100.0f, 0.0f));
	SetScale(100.0f);
	MeshComponent* mc = new MeshComponent(this);
	mc->SetMesh(game->GetRenderer()->GetMesh("Assets/Dice.gpmesh"));
	mMoveComp = new MoveComponent(this);
}

void Dice::UpdateActor(float deltaTime)
{
	Actor::UpdateActor(deltaTime);

	Vector3 axis = Vector3(1.0f, -1.0f,1.0f);
	axis.Normalize();
	float rotSpeedMax = 100.0f * Math::Pi;
	float rotSpeed = rotSpeedMax * deltaTime;
	Vector3 v = rotSpeed * axis;
	mMoveComp->SetRotSpeed(v);
}

File: StudyGamePrograming06_VB/Dice.vb
Imports OpenTK.Mathematics
Imports OpenTK.Windowing.GraphicsLibraryFramework
Public Class Dice
    Inherits Actor
    'public
    Sub New(ByRef game As Game)
        MyBase.New(game)
        SetPosition(New Vector3(200.0, -100.0, 0.0))
        SetScale(100.0)
        Dim mc As MeshComponent = New MeshComponent(Me)
        mc.SetMesh(game.GetRenderer().GetMesh("Assets/Dice.gpmesh"))
        mMoveComp = New MoveComponent(Me, 20)
    End Sub

    Public Overrides Sub UpdateActor(deltaTime As Single)
        MyBase.UpdateActor(deltaTime)
        Dim axis As Vector3 = New Vector3(1.0, 1.0, 1.0)
        axis.Normalize()
        Dim rotSpeedMax As Single = -30.0 * Math.PI
        Dim rotSpeed As Single = rotSpeedMax * deltaTime
        mMoveComp.SetRotSpeed(rotSpeed * axis)
    End Sub

    'private
    Dim mMoveComp As MoveComponent

End Class
